knack/iframeMapMessenger.js

```diff
@@ -64,61 +84,66 @@
+        records: [],
+        location: [],
       };
 
-      sendMessageToApp(markerMessage, locationViewIFrame);
+      // Request the location based on record ID
+      console.log("Requesting records for location id ", recordId);
+      $.ajax({
+        url: `https://api.knack.com/v1/scenes/scene_1039/views/view_2733/records/${recordId}`,
+        headers: headers,
+      }).then(function (res) {
+        var locationField = res["field_3300_raw"];
+        signsMarkerMessage.location = [
+          locationField.latitude,
+          locationField.longitude,
+        ];
+        console.log("requesting records for work order id ", workOrderId);
+        // Request the associated signs records
+        $.ajax({
+          url: `https://api.knack.com/v1/scenes/scene_1028/views/view_2573/records?view-work-orders-details-sign_id=${workOrderId}`,
+          headers: headers,
+        }).then(function (res) {
+          var records = res.records;
+          signsMarkerMessage.records = records;
+          sendMessageToApp(signsMarkerMessage, locationViewIFrame);
+        });
+      });
     }
-    // Location Details Page - Editable
-    $("#view_2609 #mapIFrame").on("load", function() {
-      locationDetailsMapMessage("view_2609");
-    });
-
-    // Location Details Page - Viewer
-    $("#view_2733 #mapIFrame").on("load", function() {
-      locationDetailsMapMessage("view_2733");
-    });
 
-    // Work Orders Details Page Maps
-    function workOrdersDetialsMapMessage(viewId) {
+    /**
+     * uses work order record ID to request signs and sends to iframe
+     * @param {*} viewId
+     */
+    function workOrdersDetailsMapMessage(viewId) {
       var urlArray = window.location.href.split("/");
       var recordId = urlArray[urlArray.length - 2];
-      var workOrderId = urlArray[urlArray.length - 4];
       var workOrderDetailsIFrame = $("#" + viewId + " #mapIFrame")[0]
         .contentWindow;
-
-      var markerMessage = {
-        message: "SIGNS_API_REQUEST",
-        view: myView.slice(1),
-        scene: "scene_1028",
-        token: Knack.getUserToken(),
-        app_id: Knack.application_id,
-        id: recordId,
-        workOrderId: workOrderId
-      };
-
-      sendMessageToApp(markerMessage, workOrderDetailsIFrame);
+      var headers = getHeaders(Knack.getUserToken(), Knack.application_id);
+      var view = myView.slice(1);
+
+      console.log("requesting records for work order id ", recordId);
+      $.ajax({
+        url: `https://api.knack.com/v1/scenes/scene_1028/views/${view}/records?view-work-orders-details-sign_id=${recordId}`,
+        headers: headers,
+      }).then(function (res) {
+        var records = res.records;
+        console.log("WORK ORDER SIGNS: ", records);
+        var signsMarkerMessage = {
+          message: "WORK_ORDER_SIGNS",
+          records: records,
+          location: [],
+        };
+        sendMessageToApp(signsMarkerMessage, workOrderDetailsIFrame);
+      });
```

Comment:
yes! I think that Knack data fetching in the Knack code will make this much easier to understand for the next person instead of the payload being the credentials needed to make a request for that data. 🙏